src/clj_pgp/core.clj

```diff
@@ -258,9 +259,16 @@
   "Lazily decodes a sequence of PGP objects from an input stream."
   [^InputStream input]
   (let [factory (PGPObjectFactory. input (BcKeyFingerprintCalculator.))]
-    (->>
-      (repeatedly #(.nextObject factory))
-      (take-while some?))))
+    (->> (range)
+         (map (fn [n]
+                (try
+                  (.nextObject factory)
+                  (catch Exception e
+                    (error/*handler* ::read-object-error
+                                     (.getMessage e)
+                                     (assoc (ex-data e) :stream input :nth n)
```

Comment:
`(ex-data e)` will always be nil with the code as-is. I chose to do this for consistency's sake -- as this pattern is applied elsewhere, the default is to add context to `ex-data`, not replace it.

Reply:
Should these keys be namespaced, then? If a consumer has aliased `clj-pgp.core` as `pgp` I could see `::pgp/stream` and `::pgp/index` being informative keys.